File: common/common-tenant/src/main/java/com/glowxq/tenant/business/pojo/po/Tenant.java
package com.glowxq.tenant.business.pojo.po;

import com.glowxq.core.common.entity.base.BaseEntity;
import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.io.Serial;
import java.time.LocalDateTime;

/**
* <p>
* 租户
* </p>
*
* @author glowxq
* @since 2025-06-06
*/
@Data
@Table(value = "tenant")
@Schema(description = "租户")
public class Tenant implements BaseEntity {

    @Serial
    private static final long serialVersionUID = 1L;

    @Id(keyType = KeyType.Auto)
    @Schema(description ="主键ID")
    private Long id;

    @Schema(description ="租户id")
    private String tenantId;

    @Schema(description ="租户编码，唯一标识")
    private String tenantCode;

    @Schema(description ="租户名称")
    private String tenantName;

    @Schema(description ="联系人姓名")
    private String contactName;

    @Schema(description ="联系人手机号")
    private String contactPhone;

    @Schema(description ="联系人邮箱")
    private String contactEmail;

    @Schema(description ="显示")
    private Boolean show;

    @Schema(description = "租户密码")
    private String password;

    @Schema(description ="状态：0-禁用，1-正常")
    private Boolean enable;

    @Schema(description ="到期时间")
    private LocalDateTime expiredTime;

    @Schema(description ="最大用户数")
    private Integer maxUserNum;

    @Schema(description ="当前用户数")
    private Integer currentUserNum;

    @Schema(description ="Logo地址")
    private String logoUrl;

    @Schema(description ="租户文本")
    private String text;

    @Schema(description ="系统名称")
    private String systemName;

    @Schema(description ="首页图片地址")
    private String homeImageUrl;

    @Schema(description ="主题色")
    private String themeColor;

    @Schema(description ="配置信息")
    private String config;

    @Schema(description ="自定义域名")
    private String customDomain;

    @Schema(description ="创建时间")
    private LocalDateTime createTime;

    @Schema(description ="更新时间")
    private LocalDateTime updateTime;

    @Schema(description ="是否删除")
    private String delFlag;

    @Schema(description ="创建人ID")
    private Long createId;

    @Schema(description ="更新人ID")
    private Long updateId;

    public void plusCurrentUserNum() {
        this.currentUserNum++;
    }
}

File: oj-web/src/api/interface/system/tenant/tenant.ts
export namespace ITenant {

  // 查询条件
  export interface Query {
    /** 关键字 */
    keyword?: string
    /** 租户编码 */
    tenantCode?: string
    /** 是否显示 */
    show?: boolean
    /** 是否启用 */
    enable?: boolean
    /** 页码 */
    page: number
    /** 每页条数 */
    limit: number
  }

  // 编辑form表单
  export interface Form {
    /** ID */
    id?: number
    /** 租户ID */
    tenantId?: string
    /** 租户编码 */
    tenantCode: string
    /** 租户名称 */
    tenantName: string
    /** 联系人姓名 */
    contactName?: string
    /** 联系人手机号 */
    contactPhone?: string
    /** 联系人邮箱 */
    contactEmail?: string
    /** 是否显示 */
    show?: boolean
    /** 租户密码 */
    password?: string
    /** 是否启用 */
    enable?: boolean
    /** 到期时间 */
    expiredTime?: string
    /** 最大用户数 */
    maxUserNum?: number
    /** Logo地址 */
    logoUrl?: string
    /** 租户文本 */
    text?: string
    /** 系统名称 */
    systemName?: string
    /** 首页图片地址 */
    homeImageUrl?: string
    /** 主题色 */
    themeColor?: string
    /** 配置信息 */
    config?: string
    /** 自定义域名 */
    customDomain?: string
  }

  // list或detail返回结构
  export interface Row {
    /** 主键ID */
    id: number
    /** 租户ID */
    tenantId: string
    /** 租户编码 */
    tenantCode: string
    /** 租户名称 */
    tenantName: string
    /** 联系人姓名 */
    contactName?: string
    /** 联系人手机号 */
    contactPhone?: string
    /** 联系人邮箱 */
    contactEmail?: string
    /** 是否显示 */
    show?: boolean
    /** 租户密码 */
    password?: string
    /** 是否启用 */
    enable?: boolean
    /** 到期时间 */
    expiredTime?: string
    /** 最大用户数 */
    maxUserNum?: number
    /** 当前用户数 */
    currentUserNum?: number
    /** Logo地址 */
    logoUrl?: string
    /** 租户文本 */
    text?: string
    /** 系统名称 */
    systemName?: string
    /** 首页图片地址 */
    homeImageUrl?: string
    /** 主题色 */
    themeColor?: string
    /** 配置信息 */
    config?: string
    /** 自定义域名 */
    customDomain?: string
    /** 创建时间 */
    createTime?: string
    /** 更新时间 */
    updateTime?: string
  }

}